File: app/src/main/java/com/xadaptersimple/view/RefreshView.java
package com.xadaptersimple.view;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.AppCompatImageView;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.animation.Animation;
import android.view.animation.RotateAnimation;
import android.widget.ProgressBar;

import com.xadapter.widget.XRefreshView;
import com.xadaptersimple.R;

/**
 * by y on 2017/6/21.
 */

public class RefreshView extends XRefreshView {
    private ProgressBar progressBar;
    private AppCompatTextView mText;
    private AppCompatImageView mTipsIv;

    private Animation mRotateUpAnim;
    private Animation mRotateDownAnim;

    public RefreshView(Context context) {
        super(context);
    }

    public RefreshView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public RefreshView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void initView() {
        progressBar = (ProgressBar) findViewById(com.xadapter.R.id.progressbar);
        mText = (AppCompatTextView) findViewById(com.xadapter.R.id.tv_tips);
        mTipsIv = (AppCompatImageView) findViewById(com.xadapter.R.id.iv_tips);
        mText.setText("下拉刷新");
        mText.setTextColor(ContextCompat.getColor(getContext(), R.color.colorAccent));
        initAnimation();
    }

    private void initAnimation() {
        mRotateUpAnim = new RotateAnimation(0.0f, -180.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        mRotateUpAnim.setDuration(180);
        mRotateUpAnim.setFillAfter(true);
        mRotateDownAnim = new RotateAnimation(-180.0f, 0.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f);
        mRotateDownAnim.setDuration(180);
        mRotateDownAnim.setFillAfter(true);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.layout_refresh;
    }

    @Override
    protected void onStart() {
        mTipsIv.clearAnimation();
        mTipsIv.setVisibility(VISIBLE);
        progressBar.setVisibility(INVISIBLE);
    }

    @Override
    protected void onNormal() {
        if (mState == READY) {
            mTipsIv.startAnimation(mRotateDownAnim);
        } else {
            mTipsIv.clearAnimation();
        }
        mTipsIv.setVisibility(VISIBLE);
        progressBar.setVisibility(INVISIBLE);
        mText.setText("下拉刷新");
    }

    @Override
    protected void onReady() {
        mTipsIv.startAnimation(mRotateUpAnim);
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(VISIBLE);
        mText.setText("释放立即刷新");
    }

    @Override
    protected void onRefresh() {
        progressBar.setVisibility(VISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        mText.setText("正在刷新...");
    }

    @Override
    protected void onSuccess() {
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        mTipsIv.clearAnimation();
        mText.setText("刷新成功");
    }

    @Override
    protected void onError() {
        progressBar.setVisibility(INVISIBLE);
        mTipsIv.setVisibility(INVISIBLE);
        mTipsIv.clearAnimation();
        mText.setText("刷新失败");
    }


}

File: app/src/main/java/com/xadaptersimple/view/RefreshView.kt
package com.xadaptersimple.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.view.animation.Animation
import android.view.animation.RotateAnimation
import android.widget.ProgressBar
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import com.xadapter.XRefreshView
import com.xadaptersimple.R

/**
 * by y on 2017/6/21.
 */

class RefreshView : XRefreshView {
    private lateinit var progressBar: ProgressBar
    private lateinit var mText: AppCompatTextView
    private lateinit var mTipsIv: AppCompatImageView

    private lateinit var mRotateUpAnim: Animation
    private lateinit var mRotateDownAnim: Animation

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    public override fun initView() {
        progressBar = findViewById(R.id.progressbar)
        mText = findViewById(R.id.tv_tips)
        mTipsIv = findViewById(R.id.iv_tips)
        mText.text = "下拉刷新"
        mText.setTextColor(ContextCompat.getColor(context, R.color.colorAccent))
        initAnimation()
    }

    private fun initAnimation() {
        mRotateUpAnim = RotateAnimation(0.0f, -180.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        mRotateUpAnim.duration = 180
        mRotateUpAnim.fillAfter = true
        mRotateDownAnim = RotateAnimation(-180.0f, 0.0f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
        mRotateDownAnim.duration = 180
        mRotateDownAnim.fillAfter = true
    }

    override fun getLayoutId(): Int {
        return R.layout.layout_refresh
    }

    override fun onStart() {
        mTipsIv.clearAnimation()
        mTipsIv.visibility = View.VISIBLE
        progressBar.visibility = View.INVISIBLE
    }

    override fun onNormal() {
        if (state == XRefreshView.READY) {
            mTipsIv.startAnimation(mRotateDownAnim)
        } else {
            mTipsIv.clearAnimation()
        }
        mTipsIv.visibility = View.VISIBLE
        progressBar.visibility = View.INVISIBLE
        mText.text = "下拉刷新"
    }

    override fun onReady() {
        mTipsIv.startAnimation(mRotateUpAnim)
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.VISIBLE
        mText.text = "释放立即刷新"
    }

    override fun onRefresh() {
        progressBar.visibility = View.VISIBLE
        mTipsIv.visibility = View.INVISIBLE
        mText.text = "正在刷新..."
    }

    override fun onSuccess() {
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.INVISIBLE
        mTipsIv.clearAnimation()
        mText.text = "刷新成功"
    }

    override fun onError() {
        progressBar.visibility = View.INVISIBLE
        mTipsIv.visibility = View.INVISIBLE
        mTipsIv.clearAnimation()
        mText.text = "刷新失败"
    }


}
